src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,205 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        addAll(size, list);
+    }
+
+    @SuppressWarnings("unchecked")
+    public void addAll(int index, List<T> list) {
+        checkPositionIndex(index);
+        Object[] arr = list.toArray();
+        if (arr.length == 0) {
+            return;
+        }
+        Node<T> prev;
+        Node<T> next;
+        if (index == size) {
+            next = null;
+            prev = last;
+        } else {
+            next = node(index);
+            prev = next.previous;
+        }
+        for (Object o : arr) {
+            T el = (T) o;
+            Node<T> newNode = new Node<>(prev, el, null);
+            if (prev == null) {
+                first = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            prev = newNode;
+        }
+        if (next == null) {
+            last = prev;
+        } else {
+            prev.next = next;
+            next.previous = prev;
+        }
+        size += arr.length;
     }
```

Comment:
<img width="552" alt="image" src="https://user-images.githubusercontent.com/81781479/217858894-e03457c3-bf1f-41cf-98a9-3155baa56af3.png">

it looks too similar

lets just implement addAll(list) and reuse the add(value) for it